models/Asset/Folder.php

```diff
@@ -139,7 +139,7 @@ public function getPreviewImage()
         $list->setOrder('asc');
         $list->setLimit($limit);
 
-        $totalImages = $list->getTotalCount();
+        $totalImages = min($limit, $list->getTotalCount());
```

Comment:
```suggestion
        $totalImages = $list->getCount();
```
This should have the same effect, right?